src/containers/deactivate-account-page/store/sagas.ts

```diff
@@ -0,0 +1,91 @@
+import { waitFor } from '@testing-library/dom'
+import { delay } from 'redux-saga'
+import { call, fork, put, select, takeEvery } from 'redux-saga/effects'
+
+import { ID } from 'common/models/Identifiers'
+import { getAccountUser, getUserId } from 'common/store/account/selectors'
+import { Nullable } from 'common/utils/typeUtils'
+import AudiusBackend from 'services/AudiusBackend'
+import { waitForBackendSetup } from 'store/backend/sagas'
+import { requestConfirmation } from 'store/confirmer/actions'
+import { confirmTransaction } from 'store/confirmer/sagas'
+import {
+  getConfirmCalls,
+  getConfirmLength,
+  getIsConfirming
+} from 'store/confirmer/selectors'
+import { waitForValue } from 'utils/sagaHelpers'
+import { signOut } from 'utils/signOut'
+
+import {
+  afterDeactivationSignOut,
+  deactivateAccount,
+  deactivateAccountFailed
+} from './slice'
+
+const DEACTIVATE_CONFIRMATION_UID = 'DEACTIVATE'
+
+function* handleDeactivateAccount() {
+  try {
+    yield call(waitForBackendSetup)
+    const accountUserId: Nullable<ID> = yield select(getUserId)
+    const userMetadata: ReturnType<typeof getAccountUser> = yield select(
+      getAccountUser
+    )
+    yield put(
+      requestConfirmation(
+        DEACTIVATE_CONFIRMATION_UID,
+        function* () {
+          const { blockHash, blockNumber } = yield call(
+            AudiusBackend.updateCreator,
+            { ...userMetadata, is_deactivated: true },
+            accountUserId /* note: as of writing, unused parameter */
+          )
+          const confirmed: boolean = yield call(
+            confirmTransaction,
+            blockHash,
+            blockNumber
+          )
+          if (!confirmed) {
+            throw new Error(
+              `Could not confirm account deactivation for user ${accountUserId}`
+            )
+          }
+        },
+        // @ts-ignore: confirmer is untyped
+        function* () {
+          // Do the signout in another action so confirmer can clear
+          yield put(afterDeactivationSignOut())
+        },
+        function* () {
+          yield put(deactivateAccountFailed())
+        }
+      )
+    )
+  } catch (e) {
+    console.error(e)
+    yield put(deactivateAccountFailed())
+  }
+}
+
+function* waitAndSignOut() {
+  // Wait for all confirmations to end so that reloading doesn't trigger beforeUnload dialog
+  // Note: Waiting for getIsConfirming doesn't appear to work here
```

Comment:
do you know why it didn't work?
i imagine waitForValue getIsConfirming on false should work?

Reply:
It stopped checking, unsure why. I'm pretty sure that's exactly what I tried.

Reply:
Yeah just tried again and console logged in the validator, and it only checks once and at that point returns true, then the "take" never goes again I guess and you're stuck waiting forever.